resolver/src/main/java/io/netty/resolver/HostsFileEntriesResolver.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver;
+
+import java.net.InetAddress;
+import java.util.Map;
+
+/**
+ * Resolves a hostname against the hosts file entries.
+ */
+public interface HostsFileEntriesResolver {
+
+    InetAddress getAddress(String inetHost);
+
+    /**
+     * {@link HostsFileEntriesResolver} that resolves and parses the hosts file only once.
+     */
+    enum DefaultHostsFileEntriesResolver implements HostsFileEntriesResolver {
```

Comment:
Could you:

- make this an ordinary top-level class? i.e. final class DefaultHostsFileEntriesResolver
- and add HostsFileEntriesResolver.DEFAULT which is a singleton instanceof `DefaultHostsFileEntriesResolver`.